shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -1349,4 +1505,327 @@ public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOE
+    public static final class RemoveTableKeys implements Request, WireCommand {
+
+        final WireCommandType type = WireCommandType.REMOVE_TABLE_KEYS;
+        final long requestId;
+        final String segment;
+        final String delegationToken;
+        final List<TableKey> keys;
+
+        @Override
+        public void process(RequestProcessor cp) {
+            cp.removeTableKeys(this);
+        }
+
+        @Override
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeUTF(segment);
+            out.writeUTF(delegationToken == null ? "" : delegationToken);
+            out.writeInt(keys.size());
+            for (TableKey key : keys) {
+                key.writeFields(out);
+            }
+        }
+
+        public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOException {
+            long requestId = in.readLong();
+            String segment = in.readUTF();
+            String delegationToken = in.readUTF();
+            int numberOfKeys = in.readInt();
+            List<TableKey> keys = new ArrayList<>(numberOfKeys);
+            for (int i = 0; i < numberOfKeys; i++) {
+                keys.add((TableKey) TableKey.readFrom(in, in.available()));
+            }
+            return new RemoveTableKeys(requestId, segment, delegationToken, keys);
+        }
+    }
+
+    @Data
+    public static final class TableKeysRemoved implements Reply, WireCommand {
+        final WireCommandType type = WireCommandType.TABLE_KEYS_REMOVED;
+        final long requestId;
+        final String segment;
+
+        @Override
+        public void process(ReplyProcessor cp) {
+            cp.tableKeysRemoved(this);
+        }
+
+        @Override
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeUTF(segment);
+        }
+
+        public static WireCommand readFrom(DataInput in, int length) throws IOException {
+            long requestId = in.readLong();
+            String segment = in.readUTF();
+            return new TableKeysRemoved(requestId, segment);
+        }
+    }
+
+    @Data
+    public static final class ReadTable implements Request, WireCommand {
+
+        final WireCommandType type = WireCommandType.READ_TABLE;
+        final long requestId;
+        final String segment;
+        final String delegationToken;
+        final List<TableKey> keys; // the version of the key is always set to io.pravega.segmentstore.contracts.tables.TableKey.NO_VERSION
+
+        @Override
+        public void process(RequestProcessor cp) {
+            cp.readTable(this);
+        }
+
+        @Override
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeUTF(segment);
+            out.writeUTF(delegationToken);
+            out.writeInt(keys.size());
+            for (TableKey key : keys) {
```

Comment:
See my above comment about versioning.

Reply:
Added payload size to handle future updates to TableKey.